src/main/java/com/ra4king/circuitsim/gui/peers/gates/GatePeer.java

```diff
@@ -167,8 +168,33 @@ public final void paint(GraphicsContext graphics, CircuitState circuitState) {
 				
 				graphics.fillOval(x, y, GuiUtils.BLOCK_SIZE, GuiUtils.BLOCK_SIZE);
 				graphics.strokeOval(x, y, GuiUtils.BLOCK_SIZE, GuiUtils.BLOCK_SIZE);
+			} else if (hasNegatedInput) {
+				// Imitate how a wire is drawn in Wire.paint()
+				GuiUtils.setBitColor(graphics, circuitState, c.getLinkWires());
```

Comment:
Components shouldn't be accessing PortConnection's LinkWires, plus this makes more sense to use the last pushed value of the gate:

GuiUtils.setBitColor(graphics, circuitState.getLastReceived(portConnection.getPort()));